tests/xharness/Logging/LogFile.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.IO;
+
+namespace Xharness.Logging {
+	public class LogFile : Log, ILogFile {
+		object lock_obj = new object ();
+		public string Path { get; private set; }
+		FileStream writer;
+		bool disposed;
+
+		public LogFile (ILogs logs, string description, string path, bool append = true)
+			: base (logs, description)
+		{
+			if (path == null)
+				throw new ArgumentNullException (nameof (path));
+			Path = path;
+			if (!append)
```

Comment:
>(! [](start = 6, length = 2)

Not sure this should happen in ctor really. If this blows up (e.g. path invalid or not existing) it will throw from `new` which is a bad practice.
I would rather handle that in `Write` by making the `FileMode` depend on `append` and then switch `append` after first write.